src/dotty/tools/dotc/reporting/diagnostic/messages.scala

```diff
@@ -500,4 +500,11 @@ object messages {
            |which cannot start with ${Red(illegalToken)}.""".stripMargin
     }
   }
+
+  case class ProperDefinitionNotFound()(implicit ctx: Context) extends Message(18) {
+    val kind = "Definition Not Found"
+    val msg = hl"""|A proper definition was not found in ${"@usecase"}"""
+    val explanation = ""
+  }
+  
```

Comment:
Usecases are a workaround for ugly signatures that the user of a library does not need to know about. For instance:

```
trait List[+A] {
  def map[B, That](f: A => B)(implicit bf: CanBuildFrom[List[A], B, That] : That = ???
}
```

now the users of the list are going "dafuq :confused:". Basically the implicit builder will effectively build a collection of the expected type - but the user just assumes that's what happens when you map - and doesn't need to know that.

So if we do this:

```scala
/** Map from List[A] => List[B]
  *
  * @usecase def map[B](f: A => B): List[B]
  */
def map[B, That](f: A => B)(implicit bf: CanBuildFrom[List[A], B, That]): That = ???
```

when building the documentation, the doc tool (which is a sort of compiler) will switch out the definition of `map` for the one defined in the `@usecase`. As such, the user reading the docs, won't see the ugly definition and be blissfully unaware :)

Because of this, the error occurs when the user has not written a `def` but something else (like `val map: List[B]` or w/e).

So perhaps the explanation can contain some of this information?